src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -96,6 +96,55 @@ public void addEdge(ExpressionEdge e) {
   public ExpressionGraph() {
   }
 
+  /**
+   * Adds the provided subgraph to this graph
+   * Precondition: there are no variable id conflicts between subGraph and this graph. That is
+   *               each variable identifier in the subgraph is generated WRT its scope.
+   *
+   * In the final graph, subGraphInput/subGraphOutput will no longer exist, and the edges from/to them
+   * should be from mainGraphInput/mainGraphOutput respectively.
+   *
+   * The edge from mainGraphInput to the function vertex & the edge from the function vertex to mainGraphOutput
+   * are not removed (since 1 vertex can potentially point to multiple functions), so they should be deleted by
```

Comment:
This is not true, according to my understanding; every `FunctionInvocationVertex` corresponds to exactly one function call. So it is safe to remove that vertex.